src/hdfio.cxx

```diff
@@ -2855,7 +2855,11 @@ void ReadHDF(Options &opt, vector<Particle> &Part, const Int_t nbodies,Particle
                         //self-energy
                         itemp++;
                         if (k == HDFGASTYPE) {
+			  if(opt.ihdfnameconvention != HDFSWIFTEAGLENAMES) {
                             HDF5ReadHyperSlabReal(udoublebuff,partsdatasetall[i*NHDFTYPE*NHDFDATABLOCK+k*NHDFDATABLOCK+itemp], partsdataspaceall[i*NHDFTYPE*NHDFDATABLOCK+k*NHDFDATABLOCK+itemp], 1, 1, nchunk, n, plist_id);
+			  } else {
+			    bzero(udoublebuff, chunksize * sizeof(double));
```

Comment:
`bzero` is deprecated since POSIX.1-2001; please go with `std::memset`, or at least `::memset` (I think there's a `using namespace std` in one of the headers, so if you want to access the C `memset` you'll need that `::` prefix to refer to the root namespace).